#VRML_SIM R2019a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.005098574883501008 0.9999304172226854 -0.010637915539411109 5.481821915533513
  position -3.2215138007370863 0.6157035404626513 3.7024174490920783
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
Robot {
  translation 0 0.425 0
  children [
    DEF BODY_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Box {
        size 0.5 0.1 0.2
      }
    }
    Hinge2Joint {
      jointParameters HingeJointParameters {
        anchor 0.23 0 0.135
        minStop -0.5
        maxStop 0.5
      }
      jointParameters2 JointParameters {
        minStop -3.14
        maxStop 3.14
      }
      device [
        RotationalMotor {
          name "rotational motor1"
          acceleration 8
          minPosition -0.5
          maxPosition 0.5
        }
      ]
      device2 [
        RotationalMotor {
          name "rotational motor2"
          acceleration 8
          minPosition -3.14
          maxPosition 3.14
        }
      ]
      endPoint Solid {
        translation 0.23 -0.075 0.135
        children [
          DEF LEG_SLIDE SliderJoint {
            jointParameters JointParameters {
              axis 0 1 0
              minStop -0.01
              maxStop 0.13
            }
            device [
              LinearMotor {
                name "linear motor1"
                acceleration 8
                minPosition -0.01
                maxPosition 0.13
                maxForce 65
              }
            ]
            endPoint Solid {
              translation 0 -0.2 0
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.6 0.1 0.65
                  }
                  geometry Capsule {
                    height 0.15
                    radius 0.04
                  }
                }
              ]
              name "solida"
              boundingObject Capsule {
                height 0.15
                radius 0.04
              }
              physics Physics {
              }
            }
          }
          DEF UPPER_LEG Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
            }
            geometry Capsule {
              height 0.15
              radius 0.04
            }
          }
        ]
        name "solid1"
        boundingObject USE UPPER_LEG
        physics Physics {
        }
      }
    }
    Hinge2Joint {
      jointParameters HingeJointParameters {
        anchor 0.23 0 -0.135
        minStop -0.5
        maxStop 0.5
      }
      jointParameters2 JointParameters {
        minStop -3.14
        maxStop 3.14
      }
      device [
        RotationalMotor {
          name "rotational motor3"
          acceleration 8
          minPosition -0.5
          maxPosition 0.5
        }
      ]
      device2 [
        RotationalMotor {
          name "rotational motor4"
          acceleration 8
          minPosition -3.14
          maxPosition 3.14
        }
      ]
      endPoint Solid {
        translation 0.23 -0.075 -0.135
        children [
          DEF LEG_SLIDE SliderJoint {
            jointParameters JointParameters {
              axis 0 1 0
              minStop -0.01
              maxStop 0.13
            }
            device [
              LinearMotor {
                name "linear motor2"
                acceleration 8
                minPosition -0.01
                maxPosition 0.13
                maxForce 65
              }
            ]
            endPoint Solid {
              translation 0 -0.2 0
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0 0 1
                  }
                  geometry Capsule {
                    height 0.15
                    radius 0.04
                  }
                }
              ]
              name "solida"
              boundingObject Capsule {
                height 0.15
                radius 0.04
              }
              physics Physics {
              }
            }
          }
          USE UPPER_LEG
        ]
        name "solid2"
        boundingObject USE UPPER_LEG
        physics Physics {
        }
      }
    }
    Hinge2Joint {
      jointParameters HingeJointParameters {
        anchor -0.23 0 0.135
        minStop -0.5
        maxStop 0.5
      }
      jointParameters2 JointParameters {
        minStop -3.14
        maxStop 3.14
      }
      device [
        RotationalMotor {
          name "rotational motor5"
          acceleration 8
          minPosition -0.5
          maxPosition 0.5
        }
      ]
      device2 [
        RotationalMotor {
          name "rotational motor6"
          acceleration 8
          minPosition -3.14
          maxPosition 3.14
        }
      ]
      endPoint Solid {
        translation -0.23 -0.075 0.135
        children [
          DEF LEG_SLIDE SliderJoint {
            jointParameters JointParameters {
              axis 0 1 0
              minStop -0.01
              maxStop 0.13
            }
            device [
              LinearMotor {
                name "linear motor3"
                acceleration 8
                minPosition -0.01
                maxPosition 0.13
                maxForce 65
              }
            ]
            endPoint Solid {
              translation 0 -0.2 0
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0 0 1
                  }
                  geometry Capsule {
                    height 0.15
                    radius 0.04
                  }
                }
              ]
              name "solida"
              boundingObject Capsule {
                height 0.15
                radius 0.04
              }
              physics Physics {
              }
            }
          }
          USE UPPER_LEG
        ]
        name "solid3"
        boundingObject USE UPPER_LEG
        physics Physics {
        }
      }
    }
    Hinge2Joint {
      jointParameters HingeJointParameters {
        anchor -0.23 0 0.135
        minStop -0.5
        maxStop 0.5
      }
      jointParameters2 JointParameters {
        minStop -3.14
        maxStop 3.14
      }
      device [
        RotationalMotor {
          name "rotational motor7"
          acceleration 8
          minPosition -0.5
          maxPosition 0.5
        }
      ]
      device2 [
        RotationalMotor {
          name "rotational motor8"
          acceleration 8
          minPosition -3.14
          maxPosition 3.14
        }
      ]
      endPoint Solid {
        translation -0.23 -0.075 -0.135
        children [
          DEF LEG_SLIDE SliderJoint {
            jointParameters JointParameters {
              axis 0 1 0
              minStop -0.01
              maxStop 0.13
            }
            device [
              LinearMotor {
                name "linear motor4"
                acceleration 8
                minPosition -0.01
                maxPosition 0.13
                maxForce 65
              }
            ]
            endPoint Solid {
              translation 0 -0.2 0
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0 0 1
                  }
                  geometry Capsule {
                    height 0.15
                    radius 0.04
                  }
                }
              ]
              name "solida"
              boundingObject Capsule {
                height 0.15
                radius 0.04
              }
              physics Physics {
              }
            }
          }
          USE UPPER_LEG
        ]
        name "solid4"
        boundingObject USE UPPER_LEG
        physics Physics {
        }
      }
    }
  ]
  boundingObject USE BODY_SHAPE
  physics Physics {
  }
  controller "rouqhQuadPyCtrl"
}
